contractsV5/bridges/bancor/KyberBancorReserve.sol

```diff
@@ -0,0 +1,230 @@
+
+        require(doTrade(srcToken, srcAmount, destToken, destAddress, conversionRate, validate));
+
+        return true;
+    }
+
+    event KyberNetworkSet(address kyberNetwork);
+
+    function setKyberNetwork(address _kyberNetwork) public onlyAdmin {
+        require(_kyberNetwork != address(0), "setKyberNetwork: kyberNetwork address is missing");
+
+        kyberNetwork = _kyberNetwork;
+        emit KyberNetworkSet(_kyberNetwork);
+    }
+
+    event BancorNetworkSet(address _bancorNetwork);
+    function setBancorContract(address _bancorNetwork) public onlyAdmin {
+        require(_bancorNetwork != address(0), "setBancorContract: bancorNetwork address is missing");
+
+        if (address(bancorNetwork) != address(0)) {
+            require(bancorToken.approve(address(bancorNetwork), 0), "setBancorContract: can not reset approve token");
+        }
+        bancorNetwork = IBancorNetwork(_bancorNetwork);
+        require(bancorToken.approve(address(bancorNetwork), 2 ** 255), "setBancorContract: can not approve token");
+
+        emit BancorNetworkSet(_bancorNetwork);
+    }
+
+    event FeeBpsSet(uint feeBps);
+
+    function setFeeBps(uint _feeBps) public onlyAdmin {
+        require(_feeBps < BPS, "setFeeBps: feeBps >= BPS");
+
+        feeBps = _feeBps;
+        emit FeeBpsSet(feeBps);
+    }
+
+    event TradeEnabled(bool enable);
+
+    function enableTrade() public onlyAdmin returns(bool) {
+        tradeEnabled = true;
+        emit TradeEnabled(true);
+
+        return true;
+    }
+
+    function disableTrade() public onlyAlerter returns(bool) {
+        tradeEnabled = false;
+        emit TradeEnabled(false);
+
+        return true;
+    }
+
+    function doTrade(
+        IERC20 srcToken,
+        uint srcAmount,
+        IERC20 destToken,
+        address payable destAddress,
+        uint conversionRate,
+        bool validate
+    )
+        internal
+        returns(bool)
+    {
+        // can skip validation if done at kyber network level
+        if (validate) {
+            require(conversionRate > 0);
+            if (srcToken == ETH_TOKEN_ADDRESS)
+                require(msg.value == srcAmount, "doTrade: msg value is not correct for ETH trade");
+            else
+                require(msg.value == 0, "doTrade: msg value is not correct for token trade");
+        }
+
+        if (srcToken != ETH_TOKEN_ADDRESS) {
+            // collect source amount
+            require(srcToken.transferFrom(msg.sender, address(this), srcAmount), "doTrade: collect src token failed");
+        }
+
+        IERC20[] memory path = getConversionPath(srcToken, destToken);
+        require(path.length > 0, "doTrade: couldn't find path");
+
+        // both BNT and ETH has decimals of 18 (MAX_DECIMALS)
+        uint userExpectedDestAmount = calcDstQty(srcAmount, MAX_DECIMALS, MAX_DECIMALS, conversionRate);
+        require(userExpectedDestAmount > 0, "doTrade: user expected amount must be greater than 0");
```

Comment:
please use const decimal values that are Eth and Bnt specific

Reply:
Updated